From 2497a3aa59961f94870b2e17c91d2f35da1d959d Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Fri, 9 Feb 2024 15:15:40 +0100
Subject: [PATCH 0945/1141] arm64: dts: st: move LVDS node to stm32mp255

LVDS support only on MP257 & MP255 SOCs.
Replace the compatible of lvds to "st,stm32mp25-lvds".

Change-Id: Ie90fddfef1db42f07fc8391e323945966098dc02
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/358879
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp253.dtsi | 17 -----------------
 arch/arm64/boot/dts/st/stm32mp255.dtsi | 17 +++++++++++++++++
 2 files changed, 17 insertions(+), 17 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp253.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp253.dtsi
@@ -72,18 +72,6 @@
 				status = "disabled";
 			};
 
-			lvds: lvds@48060000 {
-				#clock-cells = <0>;
-				compatible = "st,stm32-lvds";
-				reg = <0x48060000 0x2000>;
-				clocks = <&rcc CK_BUS_LVDS>, <&rcc CK_KER_LVDSPHY>;
-				clock-names = "pclk", "ref";
-				resets = <&rcc LVDS_R>;
-				feature-domains = <&rifsc STM32MP25_RIFSC_LVDS_ID>;
-				power-domains = <&CLUSTER_PD>;
-				status = "disabled";
-			};
-
 			eth2: eth2@482d0000 {
 				compatible = "st,stm32mp25-dwmac", "snps,dwmac-5.10a";
 				reg = <0x482d0000 0x4000>;
@@ -140,11 +128,6 @@
 	};
 };
 
-&ltdc {
-	clocks = <&rcc CK_BUS_LTDC>, <&rcc CK_KER_LTDC>, <&syscfg 0>, <&lvds 0>;
-	clock-names = "bus", "ref", "lcd", "lvds";
-};
-
 &optee {
 	interrupts = <GIC_PPI 15 (GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>;
 };
--- a/arch/arm64/boot/dts/st/stm32mp255.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp255.dtsi
@@ -22,6 +22,18 @@
 				status = "disabled";
 			};
 
+			lvds: lvds@48060000 {
+				#clock-cells = <0>;
+				compatible = "st,stm32mp25-lvds";
+				reg = <0x48060000 0x2000>;
+				clocks = <&rcc CK_BUS_LVDS>, <&rcc CK_KER_LVDSPHY>;
+				clock-names = "pclk", "ref";
+				resets = <&rcc LVDS_R>;
+				feature-domains = <&rifsc STM32MP25_RIFSC_LVDS_ID>;
+				power-domains = <&CLUSTER_PD>;
+				status = "disabled";
+			};
+
 			vdec: vdec@480d0000 {
 				compatible = "st,stm32mp25-vdec";
 				reg = <0x480d0000 0x3c8>;
@@ -60,3 +72,8 @@
 		};
 	};
 };
+
+&ltdc {
+	clocks = <&rcc CK_BUS_LTDC>, <&rcc CK_KER_LTDC>, <&syscfg 0>, <&lvds 0>;
+	clock-names = "bus", "ref", "lcd", "lvds";
+};
